Readme.md

```diff
@@ -34,28 +34,28 @@ Go dependencies are provided by the [Go machinery](https://nixos.org/nixpkgs/man
 
 For local development you may use `dep` to install go dependencies: `cd src/dividat-driver && dep ensure`.
 
-New Go dependencies can be added with `dep` (e.g. `dep ensure -add github.com/something/supercool`). The Nix specification of dependencies will recreated on subsequent builds (i.e. running `make`). Check in the updated `Gopkg.toml`, `Gopkg.lock` and `nix/deps.nix` files.
+New Go dependencies can be added with `dep` (e.g. `dep ensure -add github.com/something/supercool`). Make sure you run `make nix/deps.nix` to update the Nix expression containing go dependencies.
 
-### Releasing
-
-#### Building
+### Node dependencies
 
-**Currently releases can only be made from Linux.**
+Node dependencies are fetched and made available by Nix (using [`node2nix`](https://github.com/svanderburg/node2nix)).
 
-To create a release run: `make release`.
+You do not need to run `npm install`!
 
-A default environment (defined in `default.nix`) provides all necessary dependencies for building on your native system (i.e. Linux or Darwin). Running `make` will create a binary that should run on your system (at least in the default environemnt).
+To add new packages use the usual `npm` commands and run `make node-dependencies`. This will rebuild the Nix declarations for setting up node dependencies.
```

Comment:
So I open `nix-shell` to get `npm`, then I run `npm add ... && make node-dependencies`, but then I need to exit the `nix-shell` and start it again to actually get the changes. Changing a dependency goes from one step to three.

I also get these warnings when running `make node-dependencies`:

> WARNING: A lock file exists in the repository, yet it is not used in the generation process!
> As a result, the deployment of the project may fail.
> You probably want to run node2nix with the -l option to use the lock file!
>
> WARNING: There is a node_modules/ folder in the root directory of the project!
> These packages will be included in the Nix build and influence the outcome.
> If you don't want this to happen, then you should remove it before running any
> of the Nix commands!

Reply:
Yes. This is true and is an incovenience.

I think this is a trade-off between having all build-inputs being provided from a single system and easily being able to add new dependencies.